src/version_set.rs

```diff
@@ -22,38 +22,50 @@ use std::fmt::{Debug, Display};
 
 use crate::Ranges;
 
-/// Trait describing sets of versions.
+/// A set of versions.
+///
+/// See [`Ranges`] for an implementation.
+///
+/// Two version sets that contain the same versions must be equal.
+///
+/// The methods with default implementations can be overwritten for better performance, but they
+/// must be equal to the default implementation.
```

Comment:
It took me two readings to figure out what was meant by the sentence.
Perhaps "equivalent" or "functionally equivalent" instead of "equal".

Reply:
As I remember, equality was needed because version set comparisons serves as building blocks to check if sets are subsets of others in parts of the pubgrub algorithm. And as I remember it too, we have some text describing this already in some our default implementation of this?

Reply:
I've done another pass over the version set contract